sw/device/silicon_creator/lib/ownership/datatypes.h

```diff
@@ -0,0 +1,29 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_OWNERSHIP_DATATYPES_H_
+#define OPENTITAN_SW_DEVICE_SILICON_CREATOR_LIB_OWNERSHIP_DATATYPES_H_
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+typedef struct owner_key {
+  uint32_t key[16];
+} owner_key_t;
+
+typedef struct owner_signature {
+  uint32_t signature[16];
+} owner_signature_t;
+
+typedef struct owner_digest {
+  uint32_t digest[8];
```

Comment:
is this an SHA256 digest? can we just reuse the type defined in the silicon_creator hmac driver? or did you want this to be self contained?

Reply:
It is, but I wanted this to be self contained.

We should consider examining which structs have some form of external visibility and declare them in headers that have no other internal dependencies so such structs can be more easily exported.

Reply:
#22134